### ISSUE TITLE
403 Error pull image

### ISSUE THREAD

**Issue by lyrarns** [WhatsApp/WhatsApp-Business-API-Setup-Scripts#38]
Hello,

We got the following error while pulling the image:

#docker pull docker.whatsapp.biz/coreapp:v2.27.12

Trying to pull repository docker.whatsapp.biz/coreapp ...
sha256:947a1dfa3f03a282eff72f5293baa27cbdedfef6710671718b341984d5fdbac5: Pulling from docker.whatsapp.biz/coreapp
fe703b657a32: Pulling fs layer
f9df1fafd224: Downloading
a645a4b887f9: Downloading
57db7fe0b522: Waiting
0bb08bbcf873: Waiting
68e4fdb6d9f7: Waiting
a50219c293ee: Waiting
433d74e3d027: Waiting
21627e42fd2d: Waiting
9762562558a3: Waiting
299376325342: Waiting
d7cdc502f699: Waiting
0b8b567114cc: Waiting
error pulling image configuration: error parsing HTTP 403 response body: invalid character 'F' looking for beginning of value: "Forbidden!"

Basically the same error as #23 

We are in France.

**Comment by thiagocmoraes** [comment]
As you may've seem in #23 we had trouble reproducing this error. Can you provide more information on the machine trying to pull the image so we can investigate. Here's a list of things that were correlated with those errors:

* Using Windows - if that's the case, please try on a linux machine
* Some addresses in South Africa - when the same developer tried using AWS South Africa region, things worked. Are you able to try on a different machine from France? Your laptop might work. We want to pin down what conditions lead to this error. 

On top of that, some questions:
* Are you able to provide the IP address for this machine that got the error? 
* If you connect this same machine to a VPN< does it work? 
* Is there a way for us to repro the download from the same IP range you're downloading somehow?

**Comment by lyrarns** [comment]
Hello,

We tried with 2 different linux installations. Same error.
This is definitely linked to our public IP adresses. 

"If you connect this same machine to a VPN< does it work?" No, but with a shared 4G connection on a laptop it works. 

"Are you able to provide the IP address for this machine that got the error?" Yes. Will email it to you.

 "Is there a way for us to repro the download from the same IP range you're downloading somehow?" No.

**Comment by thiagocmoraes** [comment]
One update: we did check on jfrog if there was any IP blockage for this IP address provided by @lyrarns but found none. 

It's something we're still investigating but don't have a strong lead on what could cause the issue. Generally, downloading from a different IP would work around this issue.

**Comment by air-awan** [comment]
Hi @thiagocmoraes , I have similar problem.
Using a CentOS VM and Ubuntu VM with different public IP addresses yielded same error. Tried version 2.27.13 and 2.29.3, also with same error.
![image](https://user-images.githubusercontent.com/9200327/91013347-c1e40480-e611-11ea-8f0e-f33edbb724cd.png)

**Comment by thiagocmoraes** [comment]
@air-awan is there some way we could repro this from an IP in the same range? We're quite confident it's some issue around that but finding a case to repro and solve has been challenging.

**Comment by air-awan** [comment]
I filled in a feedback form in bintray site and the next day we were able to connect to the repo. I'm not sure what they did back then, but the same issue is happening again now:
```
sudo docker pull whatsapp-docker-dkr-releases.bintray.io/coreapp:v2.31.5
v2.31.5: Pulling from coreapp
2c11b7cecaa5: Pulling fs layer                                                                                          04637fa56252: Pulling fs layer                                                                                          d6e6af23a0f3: Pulling fs layer                                                                                          b4a424de92ad: Waiting                                                                                                   a4d0fc25c35d: Waiting                                                                                                   f8fb4dcc2d60: Waiting                                                                                                   a761d70d38d1: Waiting                                                                                                   0a55a2826efe: Waiting                                                                                                   27350b4f4d92: Waiting                                                                                                   b88a3f6b6dcf: Waiting                                                                                                   a3f5fef6b77c: Waiting                                                                                                   1b4039fd61d3: Waiting                                                                                                   040ec22a8b61: Waiting                                                                                                   error pulling image configuration: error parsing HTTP 403 response body: invalid character 'F' looking for beginning of value: "Forbidden!"
```

```
DEBU[2020-12-08T12:56:11.935251487Z] Calling HEAD /_ping
DEBU[2020-12-08T12:56:11.969576342Z] Calling POST /v1.40/images/create?fromImage=whatsapp-docker-dkr-releases.bintray.io%2Fcoreapp&tag=v2.31.5
DEBU[2020-12-08T12:56:11.973083988Z] hostDir: /etc/docker/certs.d/whatsapp-docker-dkr-releases.bintray.io
DEBU[2020-12-08T12:56:11.973138903Z] Trying to pull whatsapp-docker-dkr-releases.bintray.io/coreapp from https://whatsapp-docker-dkr-releases.bintray.io v2
DEBU[2020-12-08T12:56:15.321015697Z] Pulling ref from V2 registry: whatsapp-docker-dkr-releases.bintray.io/coreapp:v2.31.5
DEBU[2020-12-08T12:56:15.321968469Z] pulling blob "sha256:2c11b7cecaa5d3e2a57e290921ababbfb8572b549015168d4cbd91c340d2c566"
DEBU[2020-12-08T12:56:15.322461750Z] pulling blob "sha256:04637fa562525a9366f00e8f4b08d04a347bded1ee513738451ef9d42b4dfc4e"
DEBU[2020-12-08T12:56:15.322666829Z] pulling blob "sha256:d6e6af23a0f38c4a6511147d2a9dc06a07a7afa0669000cb62720c7eafe031ae"
ERRO[2020-12-08T12:56:17.262089913Z] Not continuing with pull after error: error pulling image configuration: error parsing HTTP 403 response body: invalid character 'F' looking for beginning of value: "Forbidden!"
DEBU[2020-12-08T12:56:17.267547051Z] pulling blob "sha256:b4a424de92ad71639c5cabadcdab0493e4067eb2f9cf109ffef40db178349238"
DEBU[2020-12-08T12:56:18.310863356Z] pulling blob "sha256:a4d0fc25c35d2b333de2088a8e918aa8d5cfaf3ae610678629e7499b3afd5e4e"
DEBU[2020-12-08T12:56:19.200567655Z] pulling blob "sha256:f8fb4dcc2d606effaf0f004bb4d8818be46816f426b6d3f7c1d55a4035d664a8"
DEBU[2020-12-08T12:56:19.319354980Z] pulling blob "sha256:a761d70d38d10bd2a0ad2609c7a6ad8846ce8d34e9c2a0176871fd4c5f6b9a9d"
```
Do you still interested to check this out @thiagocmoraes ? I can create a VM for you to test this.

**Comment by Rajamohanreddyai** [comment]
Hello lyrarns,

I was created the whole setup at that time I got the same error, but at the next second I was fixed the same. Mainly because of 2 reasons you may get the URL not found errors.

Containers are not up (db, coreapp, webapp).
You might be calling the wrong URL.
Please check the below link if do you want any clarity. https://youtu.be/hYO32rQQ9Pc

**Comment by uguracikgoz** [comment]
Hello i have been facing the same problem. I think there is a problem on container registery somehow.

**Comment by uguracikgoz** [comment]
> Hello i have been facing the same problem. I think there is a problem on container registery somehow.

Failed to pull image "docker.whatsapp.biz/web:v2.33.3": rpc error: code = Unknown desc = error pulling image configuration: error parsing HTTP 403 response body: invalid character 'F' looking for beginning of value: "Forbidden!"

**Comment by pablotdv** [comment]
The same issue here.

**Comment by ben221199** [comment]
![image](https://user-images.githubusercontent.com/12856904/121081633-425ca600-c7dd-11eb-9a3b-024b585cf9c1.png)

**Comment by nathanaelreges** [comment]
Guys I solved this problem by upgrading my docker version!
From: Docker version 19.03.13, build 4484c46d9d
To: Docker version 20.10.7, build f0df350
Does it works for you to?

**Comment by julianopoveda** [comment]
> 
> 
> Guys I solved this problem by upgrading my docker version!
> From: Docker version 19.03.13, build 4484c46d9d
> To: Docker version 20.10.7, build f0df350
> Does it works for you to?

Nope

**Comment by rogerweb** [comment]
No. I'm on same version (20.10.7) but still facing the same error.

**Comment by narender-khola** [comment]
Facing the issue on the server on which I initially had downloaded (long time back)

$ sudo  docker-compose -f prod-multiconnect-compose.yml up -d
Pulling waweb (docker.whatsapp.biz/web:v2.33.4)...
v2.33.4: Pulling from web
5d3b2c2d21bb: Pulling fs layer
3fc2062ea667: Pulling fs layer
75adf526d75b: Pulling fs layer
83b1bcf5f7c2: Waiting
52b2ced8e9f5: Waiting
36fdac3dba14: Waiting
6bb20f482041: Waiting
aece08c7edf7: Waiting
a61beca31cee: Waiting
a39badf19412: Waiting
cde6fb7c5512: Waiting
645ff2724ca8: Waiting
71c7885cdbd8: Waiting
1a53ac6cfeeb: Waiting
27388d38096a: Waiting
79a731d1785c: Waiting
7427b59d1fe7: Waiting
fb26bf746e59: Waiting
4eb39f1c025b: Waiting
6c5be97e59f3: Waiting
bcc722031b67: Waiting
39e5291c3f7a: Waiting
9173dcbad33a: Waiting
a05d1d198795: Waiting
821a900ced2b: Waiting
7616e062595d: Waiting
1ac1f78dc01d: Waiting
3a8b47c3081b: Waiting
e74bc18a6a58: Waiting
65a44e8d2760: Waiting
b14b2ce40262: Waiting
f95b4c881786: Waiting
ERROR: error pulling image configuration: error parsing HTTP 403 response body: invalid character 'F' looking for beginning of value: "Forbidden!"






$ sudo docker run hello-world
Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world
b8dfde127a29: Pull complete
Digest: sha256:9f6ad537c5132bcce57f7a0a20e317228d382c3cd61edae14650eec68b2b345c
Status: Downloaded newer image for hello-world:latest


Hello from Docker!
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
    (amd64)
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run an Ubuntu container with:
 $ docker run -it ubuntu bash

Share images, automate workflows, and more with a free Docker ID:
 https://hub.docker.com/

For more examples and ideas, visit:
 https://docs.docker.com/get-started/
